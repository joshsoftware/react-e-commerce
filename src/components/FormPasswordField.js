import React from 'react';
import FormField from './FormField';
import { setPassword } from '../actions/formActions';
import PropTypes from 'prop-types';

let password = {
  field: 'examplePassword',
  labelText: 'Password',
  type: 'password',
  name: 'password',
  placeholder: 'password'
};
const FormPasswordField = (props) => {
  password = {
    ...password,
    value: props.password,
    onChange: (evt) => {
      console.log(evt.target.value);
      props.dispatch(setPassword(evt));
    },
    invalid: props.passwordError !== null ? true : false,
    message: props.passwordError
  };
  return <FormField formfield={password} />;
};

export default FormPasswordField;

FormPasswordField.propTypes = {
  password: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  passwordError: PropTypes.string
};
