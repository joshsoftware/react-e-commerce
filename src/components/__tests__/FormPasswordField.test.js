import FormPasswordField from '../FormPasswordField';
import { render } from '@testing-library/react';
import React from 'react';

describe('FormPasswordField  Component', () => {
  let password = '';
  const dispatch = () => {};

  it('Must return FormPasswordField  Component', () => {
    const { asFragment } = render(<FormPasswordField password={password} dispatch={dispatch} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
