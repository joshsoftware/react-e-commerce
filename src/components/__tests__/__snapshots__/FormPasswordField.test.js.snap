// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPasswordField  Component Must return FormPasswordField  Component 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class=""
      for="examplePassword"
    >
      Password
    </label>
    <input
      aria-invalid="true"
      class="is-invalid form-control"
      name="password"
      placeholder="password"
      type="password"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</DocumentFragment>
`;
